projects/packages/forms/src/blocks/shared/settings/transforms.js

```diff
@@ -271,12 +257,15 @@ const transforms = {
 			blocks: [ 'jetpack/field-select' ],
 			transform: ( attributes, innerBlocks = [] ) => {
 				const optionLabels = getOptionLabelsFromInnerBlocks( innerBlocks );
+				const placeholder = attributes.placeholder ?? __( 'Select one option', 'jetpack-forms' );
 
 				return createBlock(
 					'jetpack/field-select',
 					{
 						...getFieldAttributes( attributes ),
 						options: optionLabels.length ? optionLabels : [ '' ],
+						placeholder,
+						className: placeholder ? 'has-placeholder' : '',
```

Comment:
Why is the `has-placeholder` classname being baked in here? It looks like it's usually injected in `edit.js` or dynamically in server rendering?

Reply:
It's because the field-select/edit.js won't add it - I'm not exactly sure why yet 🤔 

Previously I was hardcoding only the `placeholder` in `createTextFieldInnerBlocks()` - I though this might be neater as it's contained with a select-specific function...

Reply:
Actually, ignore this change. It's not correct. The placeholder belongs on the input field. Lemme take a closer look.

Reply:
> It's because the field-select/edit.js won't add it

I think it's because the default placeholder has to be added _after_ the block is created and assigned a clientId, or at least at the point in the lifecycle before the select field's `useSelect` is run to grab the input attributes:

`inputBlockAttributes: getBlock( clientId ).innerBlocks[ 1 ]?.attributes,`